### PR TITLE
feat: support severity threshold

### PR DESCRIPTION
**Pull request by thisislawatts**
- **refactor: rename to remove typo**
- **feat: filter findings output to remove items outside of target severity**
- **fix: load severity-threshold from configuration**
- **feat: filter output by supplied severity threshold**

### CODE REVIEW

File: internal/presenters/components.go [hunk 1]
```diff
@@ -120,11 +129,18 @@ func RenderSummary(summary *json_schemas.TestSummary, orgName string, testPath s
...
 
-	for _, severity := range summary.SeverityOrderAsc {
+	for _, severity := range reversedSlice {
 		for _, result := range summary.Results {
+			satisfyMinLevel := slices.Index(summary.SeverityOrderAsc, result.Severity) >= minLevelPointer
 			if result.Severity == severity {
+				if !satisfyMinLevel {
+					openIssueLabelledCount += renderInSeverityColor(severity, fmt.Sprintf(" %d %s ", 0, strings.ToUpper(severity)))
+					ignoredIssueLabelledCount += renderInSeverityColor(severity, fmt.Sprintf(" %d %s ", 0, strings.ToUpper(severity)))
+					continue
```
Comment (thisislawatts) [inline internal/presenters/components.go:142]: **question:** Design decision will be needed on whether we want to show `0 LOW` when in fact there were Low Issues but they have been filtered out by `--severity-threshold` setting.

**Comment by thisislawatts** [comment]
Following a review session earlier today with @PeterSchafer we agreed to refactoring https://github.com/snyk/cli/pull/5220, into GAF.

File: internal/presenters/components.go [hunk 2]
```diff
@@ -120,11 +129,18 @@ func RenderSummary(summary *json_schemas.TestSummary, orgName string, testPath s
 	openIssueLabelledCount := ""
 	ignoredIssueLabelledCount := ""
 
-	slices.Reverse(summary.SeverityOrderAsc)
+	minLevelPointer := slices.Index(summary.SeverityOrderAsc, severityMinLevel)
+	reversedSlice := reverseSlice(summary.SeverityOrderAsc)
```
[the remark this answers is not in the document]
Comment (PeterSchafer) [reply internal/presenters/components.go:133]:
```suggestion
	reversedSlice := slices.Clone(original)
	slices.Reverse(reversedSlice)
```

File: pkg/local_workflows/output_workflow.go [hunk 3]
```diff
@@ -36,13 +39,50 @@ func InitOutputWorkflow(engine workflow.Engine) error {
...
+	if !ok {
+		return nil, fmt.Errorf("invalid payload type: %T", input.GetPayload())
+	}
+	err := json.Unmarshal(payload, &summary)
+	if err != nil {
+		return input, err
+	}
+
+	minSeverity := config.GetString(configuration.FLAG_SEVERITY_THRESHOLD)
+	minSeverityLevel := slices.Index(summary.SeverityOrderAsc, minSeverity)
```
Comment (PeterSchafer) [inline pkg/local_workflows/output_workflow.go:63]: Suggestion: we currently have the same filter logic in multiple places, which makes it easy to diverge. How about changing the filtering concept a bit and thereby enable to re-use some of the logic.

1. Apply severity threshold filter on slice SeverityOrderAsc, the resulting slice contains only the Severity Levels that are above the threshold.
2. Use the FilteredSeverityOrderAsc to iterate over TestSummary
3. Use the FilteredSeverityOrderAsc to iterate over Findings

Comment (PeterSchafer) [reply pkg/local_workflows/output_workflow.go]: applied

**Review by PeterSchafer — APPROVED** [review]

**Pull request merged**